sklearn/decomposition/tests/test_kernel_pca.py

```diff
@@ -125,12 +136,31 @@ def test_kernel_pca_linear_kernel():
         np.abs(PCA(4).fit(X_fit).transform(X_pred)))
 
 
+def test_kernel_pca_linear_kernel2():
```

Comment:
There are 5 samples with 4 features, so kPCA will find 5 components while PCA will find 4 components.

Reply:
The reason originally was that for `arpack` solver in PCA we cannot select 4 components but 3. I merged the two tests, still ; you are right that it will be better for readability. I also took the opportunity to add a parametrize statement